src/main/java/seedu/trackascholar/model/major/Major.java

```diff
@@ -0,0 +1,54 @@
+package seedu.trackascholar.model.major;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.trackascholar.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Major in TrackAScholar.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidMajorName(String)}
+ */
+public class Major {
+
+    public static final String MESSAGE_CONSTRAINTS = "Majors names should be alphanumeric";
+    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+
+    public final String majorName;
```

Comment:
Please change this to major also, be careful while changing this as there is a lot of refactoring for this attribute

Reply:
Ok thank you it has been resolved.